op-batcher/batcher/sync_actions_test.go

```diff
@@ -277,6 +279,55 @@ func TestBatchSubmitter_computeSyncActions(t *testing.T) {
 			},
 			expectedLogs: happyCaseLogs,
 		},
+		{name: "localSafeL2 > safeL2, preferLocalSafeL2=false",
+			newSyncStatus: eth.SyncStatus{
+				HeadL1:      eth.BlockRef{Number: 5},
+				CurrentL1:   eth.BlockRef{Number: 2},
+				SafeL2:      eth.L2BlockRef{Number: 103, Hash: block103.Hash()},
+				LocalSafeL2: eth.L2BlockRef{Number: 104, Hash: block104.Hash()},
+				UnsafeL2:    eth.L2BlockRef{Number: 109},
+			},
+			prevCurrentL1: eth.BlockRef{Number: 1},
+			blocks:        queue.Queue[*types.Block]{},
+			channels:      []channelStatuser{},
+			expected: syncActions{
+				blocksToLoad: &inclusiveBlockRange{104, 109},
+			},
+			expectedLogs: noBlocksLogs,
+		},
+		{name: "localSafeL2 > safeL2, preferLocalSafeL2=true",
+			preferLocalSafeL2: true,
+			newSyncStatus: eth.SyncStatus{
+				HeadL1:      eth.BlockRef{Number: 5},
+				CurrentL1:   eth.BlockRef{Number: 2},
+				SafeL2:      eth.L2BlockRef{Number: 103, Hash: block103.Hash()},
+				LocalSafeL2: eth.L2BlockRef{Number: 104, Hash: block104.Hash()},
+				UnsafeL2:    eth.L2BlockRef{Number: 109},
+			},
+			prevCurrentL1: eth.BlockRef{Number: 1},
+			blocks:        queue.Queue[*types.Block]{},
+			channels:      []channelStatuser{},
+			expected: syncActions{
+				blocksToLoad: &inclusiveBlockRange{105, 109},
+			},
+			expectedLogs: noBlocksLogs,
+		},
+		{name: "LocalSafeL2=0,SafeL2>0", // This shouldn't ever happen, but has occurred due to bugs
+			newSyncStatus: eth.SyncStatus{
+				HeadL1:      eth.BlockRef{Number: 5},
+				CurrentL1:   eth.BlockRef{Number: 2},
+				SafeL2:      eth.L2BlockRef{Number: 104, Hash: block103.Hash()},
```

Comment:
wondering why number and hash mismatch here? Non-essential for the test, but a bit confusing if irrelevant.
```suggestion
				SafeL2:      eth.L2BlockRef{Number: 104, Hash: block104.Hash()},
```